src/Core/Injector/InjectorNotFoundException.php

```diff
@@ -3,7 +3,14 @@
 namespace SilverStripe\Core\Injector;
 
 use Psr\Container\NotFoundExceptionInterface;
+use Throwable;
 
 class InjectorNotFoundException extends \Exception implements NotFoundExceptionInterface
 {
+    public function __construct($message = "", $code = 0, Throwable $previous = null)
```

Comment:
I wonder if it would be better to expose an accessor method on this class, e.g. `getMissingClassName()` that returned the FQCN, rather than forcing callers to rely on the string format of the class-not-fund message.

The class name could be a 4th argument to the constructor, defaulting to null, and you could try on string parsing only in those cases when the argument was missed.

InjectionCreator and Injector::get could pass the $class argument directly.

Reply:
The only place in core this exception seems to be caught is in `ImageManipulation`, and it doesn't try to do anything with the class name, so this might be a little overkill?

Reply:
Perhaps in that case it's better to use regexes in our tests to pass on the presence of an optional "?

Reply:
I'm not sure that "InjectorNotFoundExpection magically removes quotes from any message passed to it" is a necessary part of our API.

Reply:
Yep that's sensible, have dropped this out and added regex to the affected tests.